### PR TITLE
Fixed getting error on Windows with cloud provider

### PR DESCRIPTION
**Pull request by thxCode**
**Problem:**
- Could not deploy workloads when using Azure cloud provider
- Could not use GCE disk when using GCE cloud provider

**Solution:**
- We need to indicate the real node-ip for Windows kubelet
- Correct the right path to put down the Get-GCEPdName.dll

**Issue:**
https://github.com/rancher/rancher/issues/22612
https://github.com/rancher/rancher/issues/20839

### CODE REVIEW

**Comment by orangedeng** [comment]
LGTM

File: windows/cloud-provider.psm1 [hunk 1]
```diff
@@ -111,4 +111,91 @@ function Complete-AzureCloudConfig
...
+
+    if ($CloudProviderName) {
+        try {
+            # repair contain route for `169.254.169.254` when using cloud provider
+            $actualGateway = $(route.exe PRINT 0.0.0.0 | Where-Object {$_ -match '0\.0\.0\.0.*[a-z]'} | Select-Object -First 1 | ForEach-Object {($_ -replace '0\.0\.0\.0|[a-z]|\s+',' ').Trim() -split ' '} | Select-Object -First 1)
+            $expectedGateway = $(route.exe PRINT 169.254.169.254 | Where-Object {$_ -match '169\.254\.169\.254'} | Select-Object -First 1 | ForEach-Object {($_ -replace '169\.254\.169\.254|255\.255\.255\.255|[a-z]|\s+',' ').Trim() -split ' '} | Select-Object -First 1)
+            if ($actualGateway -ne $expectedGateway) {
+                route.exe ADD 169.254.169.254 MASK 255.255.255.255 $actualGateway METRIC 1 | Out-Null
+            }
+        } catch {}
```
Comment (alena1108) [inline windows/cloud-provider.psm1]: don't know powershell that much, but do we need to log anything in the catch block?

File: windows/cloud-provider.psm1 [hunk 2]
```diff
@@ -111,4 +111,91 @@ function Complete-AzureCloudConfig
...
+        
+        switch -Regex ($CloudProviderName) {
+            '^\s*aws\s*$' {
+                # gain private DNS name
+                $nodeName = $(curl.exe -s "http://169.254.169.254/latest/meta-data/hostname")
+                if (-not $?) {
+                    Log-Error "Failed to gain the priave DNS name for AWS instance: $nodeName"
```
Comment (alena1108) [inline windows/cloud-provider.psm1]: would the exact error be logged too?

File: windows/cloud-provider.psm1 (hunk 1 above)
Comment (thxCode) [reply windows/cloud-provider.psm1]: I will submit another PR to complete this.

**Pull request merged**